### ISSUE TITLE
Reloading Via React Native Debug Menu Causes Crash with Turbo Modules in Bare

### ISSUE THREAD

**Issue by ksitko** [expo/expo#10389]
### Summary of Issue

When Turbo Modules are enabled and Reanimated V2 is used, attempting to do a full reload in either iOS sim or Device causes a crash with a `com.facebook.react.JavaScript (13): signal SIGABRT` error in JSCRuntime::~JSCRuntime. Fast refresh does not cause a similar issue.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

```
Expo CLI 3.27.10 environment info:
    System:
      OS: macOS 10.15.6
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 14.4.0 - /usr/local/bin/node
      Yarn: 1.22.5 - ~/.yarn/bin/yarn
      npm: 6.14.8 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.9.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.0, DriverKit 19.0, macOS 10.15, tvOS 14.0, watchOS 7.0
    IDEs:
      Android Studio: 3.6 AI-192.7142.36.36.6392135
      Xcode: 12.0.1/12A7300 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^39.0.0 => 39.0.2 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: 0.63.2 => 0.63.2 
      react-native-web: ~0.13.7 => 0.13.13 
    Expo Workflow: bare
```

### Reproducible Demo

Error is in Bare workflow and Reanimated V2 is currently not supported by Snack.
 
### Steps to Reproduce

Enable Turbomodules in ejected Bare workflow expo app and follow Reanimated V2 documentation to change AppDelegate.mm.

### Expected Behavior vs Actual Behavior

Expected to simply reload.

**Comment by brentvatne** [comment]
hi there, you might want to post this to the reanimated 2 issues!

**Comment by ksitko** [comment]
@brentvatne Apologies, I wasn't sure if it was a reanimated v2 specific or expo. Reanimated V2 has [fixed this](https://github.com/software-mansion/react-native-reanimated/pull/1256) and the fix is included in `2.0.0-alpha.7`. It would be nice if the [docs could be updated](https://docs.expo.io/versions/latest/sdk/reanimated/) to recommend that version.
